api/src/main/java/org/pmiops/workbench/institution/InstitutionUtils.java

```diff
@@ -0,0 +1,51 @@
+package org.pmiops.workbench.institution;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.pmiops.workbench.model.Institution;
+import org.pmiops.workbench.model.InstitutionTierRequirement;
+
+/** Utilities for RW institution related functionalities. */
+public class InstitutionUtils {
+  private InstitutionUtils() {}
+
+  /**
+   * Finds the list of email addresses which from a {@link Institution}'s tier requirement by given
+   * tier. Returns empty result if access tier is not found in tier requirement.
+   */
+  public static Set<String> getEmailAddressesByTierOrEmptySet(
+      Institution institution, String accessTierShortName) {
+    return institution.getTierEmailAddresses().stream()
+        .filter(t -> t.getAccessTierShortName().equals(accessTierShortName))
+        .flatMap(
+            tea ->
+                tea.getEmailAddresses() == null ? Stream.empty() : tea.getEmailAddresses().stream())
+        .collect(Collectors.toSet());
+  }
+
+  /**
+   * Finds the list of email domains which from a {@link Institution}'s tier requirement by given
+   * tier. Returns empty result if access tier is not found in tier requirement.
+   */
+  public static Set<String> getEmailDomainsByTierOrEmptySet(
+      Institution institution, String accessTierShortName) {
+    return institution.getTierEmailDomains().stream()
+        .filter(t -> t.getAccessTierShortName().equals(accessTierShortName))
+        .flatMap(
+            tea -> tea.getEmailDomains() == null ? Stream.empty() : tea.getEmailDomains().stream())
```

Comment:
nit: `tea` as a variable name it seems was a copy paste, probably should be `ted` instead now.

But also, I would just use `t` for consistency with the above filter